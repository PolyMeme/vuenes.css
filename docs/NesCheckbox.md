# nes-checkbox 

Vue Component implementation of the Nes Checkbox 

- **author** - Jesus Fernando Alvarez Franco 
- **license** - MIT 

## slots 

- `default` Default Slot used for the Inside content of the Checkbox 

## props 

- `checked` ***Boolean*** (*optional*) `default: false` 

  Define if the checkbox is checked on init 

- `dark` ***Boolean*** (*optional*) `default: false` 

  Define if the checkbox is dark. Will use the `is-dark` class 

## computed properties 

- `checkboxClasses` 

  Computed Property that uses the checkbox classes depending of the props used. 

   **dependencies:** `dark` 


## events 

- `select` 

  Fired when someone selects the checkbox 

## methods 

- `click()` 

  Method that emits the `select` event 

