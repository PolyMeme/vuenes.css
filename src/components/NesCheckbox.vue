<template>
  <label>
    <input
      type="checkbox"
      class="nes-checkbox"
      :class="checkboxClasses"
      :checked="checked"
      @click="click"
    >
    <span>
      <!-- Default Slot used for the Inside content of the Checkbox -->
      <slot />
    </span>
  </label>
</template>

<script>
/**
 * Vue Component implementation of the Nes Checkbox
 * @author Jesus Fernando Alvarez Franco
 * @license MIT
 */
export default {
  props: {
    /**
     * Define if the checkbox is checked on init
     * @default false
     */
    checked: {
      type: Boolean,
      default: false
    },
    /**
     * Define if the checkbox is dark. Will use the `is-dark` class
     * @default false
     */
    dark: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    /**
     * Computed Property that uses the checkbox classes depending of the props used.
     * @returns {{'is-dark': BooleanConstructor}}
     */
    checkboxClasses () {
      return {
        'is-dark': this.dark
      }
    }
  },
  methods: {
    /**
     * Method that emits the `select` event
     */
    click () {
      /**
       * Fired when someone selects the checkbox
       */
      this.$emit('select')
    }
  }
}
</script>
