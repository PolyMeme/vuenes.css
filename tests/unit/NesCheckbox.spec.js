import { mount } from '@vue/test-utils'
import NesCheckbox from '../../src/components/NesCheckbox.vue'

describe('NesCheckbox.vue', () => {
  const createCheckboxWrapper = (propsData = {}, slot) => {
    return mount(NesCheckbox, {
      propsData,
      slots: {
        default: slot || 'I am a checkbox!'
      }
    })
  }

  it('should render the checkbox properly', () => {
    const wrapper = createCheckboxWrapper()

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.find('label>input[type="checkbox"].nes-checkbox').exists()).toBe(true)
    expect(wrapper.text()).toBe('I am a checkbox!')
  })

  it('should emit select event on click', () => {
    const wrapper = createCheckboxWrapper()

    // Trigger Click and Emit Three Times
    const selector = 'label>input[type="checkbox"].nes-checkbox'
    wrapper.find(selector).trigger('click')
    wrapper.find(selector).trigger('click')
    wrapper.find(selector).trigger('click')

    expect(wrapper.emitted('select')).toBeTruthy()
    expect(wrapper.emitted('select').length).toBe(3)
  })

  it('should print html inside slot', () => {
    const wrapper = createCheckboxWrapper({}, '<h2>Hello there!</h2>')
    const slot = wrapper.find('span')

    expect(wrapper.exists()).toBe(true)
    expect(slot.exists()).toBe(true)
    expect(slot.html()).toBe('<span><h2>Hello there!</h2></span>')
  })

  it('should match snapshot', () => {
    const wrapper = createCheckboxWrapper({}, '<h2>Hello there!</h2>')

    expect(wrapper.html()).toMatchSnapshot()
  })
})
