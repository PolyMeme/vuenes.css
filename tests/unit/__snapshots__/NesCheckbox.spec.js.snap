// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NesCheckbox.vue should match snapshot 1`] = `<label><input type="checkbox" class="checkbox"> <span><h2>Hello there!</h2></span></label>`;
